### PR TITLE
sushiStrategy - hardhart mainnet - fork tests (WIP)

### LINKED ISSUE #14: Test Coverage for Sushi Hodl
Likely need mainnet fork to test? I'm guessing

### LINKED ISSUE #65: Add task to update strategy on vault
Announce and then 12 hours later task to complete update to new stgrategy:

- first task calls `  function announceStrategyUpdate(address _strategy)`
- after 12 hours the second task can call `function setStrategy(address _strategy)`

### PR DESCRIPTION
**Pull request by defistar**
**List of changes:**

- sol compiler version upgraded from 0.8.0 -> 0.8.4
- hardhat fork tests for MasterChefHodlStrategy

### CODE REVIEW

File: test/strategies/sushiHODL/MasterChefHodlStrategyTest.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    [user] = await ethers.getSigners();
+
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [strategyOwner]}
+    );
+    
+  });
+
+  it("masterChefHodlStrategy should have WMatic routes initialised", async () => {
```
Comment (NoahMarconi) [inline test/strategies/sushiHODL/MasterChefHodlStrategyTest.js]: can you phrase as `it should ....`

if you want to prefix all with `masterChefHodlStrategy` you can add a describe block and pass the string there. E.g.

```
describe("MasterChefV2 - USDC_USDT mainnet fork Tests", function () {
    describe("masterChefHodlStrategy", function () {
        it("should....");

    });'

...
```

File: test/strategies/sushiHODL/MasterChefHodlStrategyTest.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const { ethers, upgrades } = require("hardhat");
```
Comment (NoahMarconi) [inline test/strategies/sushiHODL/MasterChefHodlStrategyTest.js]: Can we do typescript tests?

**Review by NoahMarconi — CHANGES_REQUESTED** [review]

File: .env-example [hunk 3]
```diff
@@ -1,5 +1,38 @@
 INFURA_PROJECT_ID=
 MATICVIGIL_APP_ID=
-DEPLOYER_PRIVATE_KEY=
+RINKEBY_ADMIN=
+RINKEBY_PRIVATE_KEY=
+KOVAN_ADMIN=
```
Comment (NoahMarconi) [inline .env-example]: Where did all these come from? Can we delete again

**Review by NoahMarconi — CHANGES_REQUESTED** [review]

File: test/strategies/sushiHODL/MasterChefHodlStrategyTest.js (hunk 2 above)
Comment (defistar) [reply test/strategies/sushiHODL/MasterChefHodlStrategyTest.js]: @NoahMarconi will update to typescript

File: .env-example (hunk 3 above)
Comment (defistar) [reply .env-example]: these are the properties from .env
will update .env to retain only those which are being used
@NoahMarconi

**Review by NoahMarconi — APPROVED** [review]

**Pull request merged**